pkg/controller/build/osbuildcontroller.go

```diff
@@ -0,0 +1,280 @@
+		kind = "<unknown object kind>"
+	}
+
+	name := e.kubeObject.GetName()
+
+	if e.funcName == "" {
+		return fmt.Sprintf("<kind: %q, name: %q>", kind, name)
+	}
+
+	return fmt.Sprintf("<kind: %q, name: %q, func: %q>", kind, name, e.getFuncName())
+}
+
+func (ctrl *OSBuildController) enqueueFuncForObject(obj kubeObject, toRun func(context.Context) error) {
+	info := &enqueuedFuncInfo{
+		kubeObject: obj,
+	}
+
+	// Add the callers name to the id so that if / when something is dropped out
+	// of the queue, it's more obvious what was happening at the time.
+	pc, _, _, ok := runtime.Caller(1)
+	details := runtime.FuncForPC(pc)
+	if ok && details != nil {
+		info.funcName = details.Name()
+	}
+
+	ctrl.execQueue.EnqueueAfterWithNameAndContext(ctrl.ctx, info.String(), ctrl.config.UpdateDelay, func(ctx context.Context) error {
+		cctx, cancel := context.WithCancel(ctx)
+		defer cancel()
+		return toRun(cctx)
+	})
+}
+
+func (ctrl *OSBuildController) addMachineOSBuild(cur interface{}) {
+	mosb := cur.(*mcfgv1alpha1.MachineOSBuild)
+	ctrl.enqueueFuncForObject(mosb, func(ctx context.Context) error {
+		return ctrl.buildReconciler.AddMachineOSBuild(ctx, mosb)
+	})
+}
+
+func (ctrl *OSBuildController) updateMachineOSBuild(old, cur interface{}) {
+	oldMOSB := old.(*mcfgv1alpha1.MachineOSBuild)
+	curMOSB := cur.(*mcfgv1alpha1.MachineOSBuild)
+	ctrl.enqueueFuncForObject(curMOSB, func(ctx context.Context) error {
+		return ctrl.buildReconciler.UpdateMachineOSBuild(ctx, oldMOSB, curMOSB)
+	})
+}
+
+func (ctrl *OSBuildController) deleteMachineOSBuild(cur interface{}) {
+	mosb := cur.(*mcfgv1alpha1.MachineOSBuild)
+	ctrl.enqueueFuncForObject(mosb, func(ctx context.Context) error {
+		return ctrl.buildReconciler.DeleteMachineOSBuild(ctx, mosb)
+	})
+}
+
+func (ctrl *OSBuildController) addPod(cur interface{}) {
+	pod := cur.(*corev1.Pod)
+	ctrl.enqueueFuncForObject(pod, func(ctx context.Context) error {
+		return ctrl.buildReconciler.AddPod(ctx, pod)
+	})
+}
+
+func (ctrl *OSBuildController) updatePod(old, cur interface{}) {
+	oldPod := old.(*corev1.Pod)
+	curPod := cur.(*corev1.Pod)
+
+	ctrl.enqueueFuncForObject(curPod, func(ctx context.Context) error {
+		return ctrl.buildReconciler.UpdatePod(ctx, oldPod, curPod)
+	})
+}
+
+func (ctrl *OSBuildController) addMachineOSConfig(newMOSC interface{}) {
+	m := newMOSC.(*mcfgv1alpha1.MachineOSConfig).DeepCopy()
+	ctrl.enqueueFuncForObject(m, func(ctx context.Context) error {
+		return ctrl.buildReconciler.AddMachineOSConfig(ctx, m)
+	})
+}
+
+func (ctrl *OSBuildController) updateMachineOSConfig(old, cur interface{}) {
+	oldMOSC := old.(*mcfgv1alpha1.MachineOSConfig).DeepCopy()
+	curMOSC := cur.(*mcfgv1alpha1.MachineOSConfig).DeepCopy()
+	ctrl.enqueueFuncForObject(curMOSC, func(ctx context.Context) error {
+		return ctrl.buildReconciler.UpdateMachineOSConfig(ctx, oldMOSC, curMOSC)
+	})
+}
+
+func (ctrl *OSBuildController) deleteMachineOSConfig(cur interface{}) {
+	mosc, ok := cur.(*mcfgv1alpha1.MachineOSConfig)
```

Comment:
Note to reviewer: I see this pattern in other controllers within the MCO for deletion. I'm not sure how necessary it is, but I would at least like to get confirmation that it is either necessary or that it is not necessary.